core/controllers/topic_viewer.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the topic viewer page."""
+
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import story_services
+from core.domain import topic_services
+import feconf
+
+
+class TopicPageDataHandler(base.BaseHandler):
+    """Manages the data that needs to be displayed to a learner on the topic
+    viewer page.
+    """
+
+    @acl_decorators.can_access_topic_viewer_page
+    def get(self, topic_name):
+        """Handles GET requests."""
+
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        topic = topic_services.get_topic_by_name(topic_name)
+        canonical_stories = [
+            story_services.get_story_by_id(
+                canonical_story_id) for canonical_story_id
+            in topic.canonical_story_ids]
+        additional_stories = [
+            story_services.get_story_by_id(
+                additional_story_id) for additional_story_id
+            in topic.additional_story_ids]
+
+        canonical_story_dicts = [{
+            'id': story.id,
```

Comment:
This should probably be a method on the Story object like story.to_summary_dict() -- it's a bit too in-depth for the controller layer.

Speaking of which, shouldn't you be getting the story summaries instead?

Reply:
Actually no, we need stories' titles and description to display on the topic viewer page. Story summary doesn't provide with the description and for each story.
We need just the info regarding the topic and fetching for stories would mean more calls (for stories' summary + stories' description). I found this way better.

Reply:
In that case I think we should modify the definition of the summary object/model (and how it is computed) so that it includes everything that is needed to actually display the summary.

Reply:
Okay done.